lib/puppet/provider/rds_instance/v2.rb

```diff
@@ -0,0 +1,122 @@
+require_relative '../../../puppet_x/puppetlabs/aws.rb'
+
+Puppet::Type.type(:rds_instance).provide(:v2, :parent => PuppetX::Puppetlabs::Aws) do
+  confine feature: :aws
+
+  mk_resource_methods
+
+  def self.instances
+    regions.collect do |region|
+      instances = []
+      rds_client(region).describe_db_instances.each do |response|
+        response.data.db_instances.each do |db|
+          unless db.db_instance_status =~ /^deleted$|^deleting$/
+            hash = db_instance_to_hash(region, db)
+            instances << new(hash) if hash[:name]
+          end
+        end
+      end
+      instances
+    end.flatten
+  end
+
+  read_only(:auto_minor_version_upgrade,
+    :backup_retention_period, :character_set_name, :creation_date_time,
+    :iops, :master_username,
+    :multi_az, :backup_window, :vpc_id, :license_model)
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name] # rubocop:disable Lint/AssignmentInCondition
+        resource.provider = prov if resource[:region] == prov.region
+      end
+    end
+  end
+
+  def self.db_instance_to_hash(region, instance)
+    if instance.respond_to?('skip_final_snapshot')
+      skip_final_snapshot = instance.skip_final_snapshot
+    else
+      skip_final_snapshot = true
+    end
+    if instance.respond_to?('final_db_snapshot_identifier')
+      final_db_snapshot_identifier = instance.final_db_snapshot_identifier
+    else
+      final_db_snapshot_identifier = ''
+    end
+    if instance.respond_to?('backup_retention_period')
+      backup_retention_period = instance.backup_retention_period
+    else
+      backup_retention_period = 0
+    end
+    config = {
+      ensure: :present,
+      name: instance.db_instance_identifier,
+      region: region,
+      engine: instance.engine,
+      db_instance_class: instance.db_instance_class,
+      master_username: instance.master_username,
+      db_name: instance.db_name,
+      allocated_storage: instance.allocated_storage,
+      storage_type: instance.storage_type,
+      license_model: instance.license_model,
+      multi_az: instance.multi_az,
+      iops: instance.iops,
+      backup_retention_period: backup_retention_period,
+      skip_final_snapshot: skip_final_snapshot,
+      final_db_snapshot_identifier: final_db_snapshot_identifier,
+      db_parameter_group_name: instance.db_parameter_groups.collect(&:db_parameter_group_name).first,
+      db_security_groups: instance.db_security_groups.collect(&:db_security_group_name),
+    }
+    if instance.respond_to?('endpoint')
```

Comment:
This doesn't check for endpoint being nil. I just hit in this in the after part of the acceptance tests.